Sorting Auditlog by actor causes 500 error

### Describe the bug

When testing the Auditlog tool, I discovered that sorting the table by "Actor" results in a 500 error. 

#### Console output
```
GET http://localhost:8001/api/1/auditlog/?page=1&search=&page_size=10&ordering=actor&_=1760084095777 500 (Internal Server Error)
```

#### Network tab
```
FieldError at /api/1/auditlog/
Field 'actor' does not generate an automatic reverse relation and therefore cannot be used for reverse querying. If it is a GenericForeignKey, consider adding a GenericRelation.
```

### To Reproduce

1. Open the "Auditlog" tool
2. Try to sort the table by "Actor"
3. You should get an an alert about an "Ajax error"

### Expected behavior

The table should sort without issue.

### Screenshots

<img width="504" height="226" alt="Image" src="https://github.com/user-attachments/assets/5725728f-47bf-4818-bfe2-3aac2ae95fea" />

### Tracebacks

```pytb
[Fri Oct 10 10:18:12 2025] [ERROR] [pid=23979 django.request] Internal Server Error: /api/1/auditlog/
Traceback (most recent call last):
  File "/home/vscode/.venv/lib/python3.11/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/home/vscode/.venv/lib/python3.11/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vscode/.venv/lib/python3.11/site-packages/django/views/decorators/csrf.py", line 56, in wrapper_view
    return view_func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vscode/.venv/lib/python3.11/site-packages/rest_framework/viewsets.py", line 125, in view
    return self.dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vscode/.venv/lib/python3.11/site-packages/rest_framework/views.py", line 515, in dispatch
    response = self.handle_exception(exc)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vscode/.venv/lib/python3.11/site-packages/rest_framework/views.py", line 475, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/home/vscode/.venv/lib/python3.11/site-packages/rest_framework/views.py", line 486, in raise_uncaught_exception
    raise exc
  File "/home/vscode/.venv/lib/python3.11/site-packages/rest_framework/views.py", line 512, in dispatch
    response = handler(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vscode/.venv/lib/python3.11/site-packages/rest_framework/mixins.py", line 38, in list
    queryset = self.filter_queryset(self.get_queryset())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vscode/.venv/lib/python3.11/site-packages/rest_framework/generics.py", line 154, in filter_queryset
    queryset = backend().filter_queryset(self.request, queryset, self)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vscode/.venv/lib/python3.11/site-packages/rest_framework/filters.py", line 327, in filter_queryset
    return queryset.order_by(*ordering)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vscode/.venv/lib/python3.11/site-packages/django/db/models/query.py", line 1659, in order_by
    obj.query.add_ordering(*field_names)
  File "/home/vscode/.venv/lib/python3.11/site-packages/django/db/models/sql/query.py", line 2222, in add_ordering
    self.names_to_path(item.split(LOOKUP_SEP), self.model._meta)
  File "/home/vscode/.venv/lib/python3.11/site-packages/django/db/models/sql/query.py", line 1701, in names_to_path
    raise FieldError(
django.core.exceptions.FieldError: Field 'actor' does not generate an automatic reverse relation and therefore cannot be used for reverse querying. If it is a GenericForeignKey, consider adding a GenericRelation.
[10/Oct/2025 10:18:12] "GET /api/1/auditlog/?page=1&search=&page_size=10&ordering=actor&_=1760084095779 HTTP/1.1" 500 136292
```

### Environment (please complete the following information):

<!--
 - If the problem is with the web ui: Your OS, Browser and browser version [e.g. Windows, Chrome 64]
 - OS on the NAV server [e.g. Debian 10]
 - NAV version installed [e.g. 4.9.8]
 - Method of installation [e.g. from Debian package]
-->
* MacOS, Brave 1.83
* Run in local development, from master branch. Also present in 5.14.x and 5.13.x.

### Additional context

<!-- Add any other context about the problem here. -->

--- a/changelog.d/3581.fixed.md
+++ b/changelog.d/3581.fixed.md
@@ -1,0 +1,1 @@
+Reenable sorting by actor in auditlog table, now actually working robustly!

--- a/python/nav/auditlog/api.py
+++ b/python/nav/auditlog/api.py
@@ -13,16 +13,17 @@
 # more details.  You should have received a copy of the GNU General Public
 # License along with NAV. If not, see <http://www.gnu.org/licenses/>.
 
-import operator
+from django.db.models import Case, F, IntegerField, OuterRef, Q, Subquery, When
+from django.db.models.functions import Cast
 
-from django.db.models import Q
+from django_filters.rest_framework import DjangoFilterBackend
 
-from rest_framework import serializers
-from rest_framework import viewsets, filters
+from rest_framework import filters, serializers, viewsets
 
 from nav.web.api.v1.views import NAVAPIMixin
 
 from nav.models.manage import Interface
+from nav.models.profiles import Account
 
 from .models import LogEntry
 
@@ -75,20 +76,33 @@ class MultipleFilter(filters.BaseFilterBackend):
         return queryset
 
 
-class CustomOrderingFilter(filters.BaseFilterBackend):
-    def filter_queryset(self, request, queryset, view):
-        """Order by actor login
+class LogEntryOrderingFilter(filters.OrderingFilter):
+    """Custom ordering filter that handles the 'actor' field specially.
 
-        Sad things happen if the actor is not an account
-        """
+    The 'actor' field is a LegacyGenericForeignKey and cannot be used with
+    Django's queryset order_by(). Instead, we use a subquery to fetch the
+    actor's login from the Account table for efficient database-level sorting.
+    """
+
+    def filter_queryset(self, request, queryset, view):
         ordering = request.query_params.get('ordering')
         if ordering in ['-actor', 'actor']:
-            return sorted(
-                queryset,
-                key=operator.attrgetter('actor.login'),
-                reverse=ordering.startswith('-'),
+            actor_login = Case(
+                When(
+                    actor_pk__regex=r'^\d+$',
+                    then=Subquery(
+                        Account.objects.filter(
+                            id=Cast(OuterRef('actor_pk'), IntegerField())
+                        ).values('login')[:1]
+                    ),
+                ),
+                default=F('actor_pk'),
             )
-        return queryset
+            queryset = queryset.annotate(actor_login=actor_login)
+            if ordering == '-actor':
+                return queryset.order_by(F('actor_login').desc(nulls_last=True))
+            return queryset.order_by(F('actor_login').asc(nulls_last=True))
+        return super().filter_queryset(request, queryset, view)
 
 
 class NetboxFilter(filters.BaseFilterBackend):
@@ -130,9 +144,11 @@ class LogEntryViewSet(NAVDefaultsMixin, viewsets.ReadOnlyModelViewSet):
 
     Logentries are created behind the scenes by the subsystems themselves."""
 
-    filter_backends = NAVDefaultsMixin.filter_backends + (
+    filter_backends = (
+        filters.SearchFilter,
+        DjangoFilterBackend,
+        LogEntryOrderingFilter,
         MultipleFilter,
-        CustomOrderingFilter,
         NetboxFilter,
     )
     queryset = LogEntry.objects.all()

--- a/python/nav/web/templates/auditlog/_logentry_list.html
+++ b/python/nav/web/templates/auditlog/_logentry_list.html
@@ -87,7 +87,6 @@
              },
              {
                  data: columns[1],
-                 orderable: false,
                  render: escapeHtml
              },
              { data: columns[2] },

